tutorials/2d/2d_lights_and_shadows.rst

```diff
@@ -0,0 +1,226 @@
+:ref:`LightOccluder2D's <class_LightOccluder2D>` will treat the position of the light as the center of the :ref:`Light <class_Light2D>` 
+node which is why we want the blob to be centered on its parent :ref:`Light <class_Light2D>`.
+
+.. image:: img/light_shadow_all_lights.png
+
+.. note:: At the time of writing this tutorial the stable version is 3.0 but the master version that will become 3.1
+          contains many changes to the animation system. So the animations contained in the demo will not be covered.
+
+Right now the scene should look too bright this is because all three lights are adding color to the scene. 
+This is why the demo uses a :ref:`CanvasModulate <class_CanvasModulate>` in the scene. The
+:ref:`CanvasModulate <class_CanvasModulate>` multiples the entire viewport by a specific color.
+
+Add a :ref:`CanvasModulate <class_CanvasModulate>` to the scene and set its color to ``rgb(70, 70, 70)``. 
+This will make the scene sufficiently dark to see the effects of the lights distinctly.
+
+.. image:: img/light_shadow_ambient.png
+
+Now we add the shadow casters.
+
+The demo uses a :ref:`Node <class_Node2D>` named "casters" to organize the shadow casters. Add a 
+:ref:`Node2D <class_Node2D>` to the scene. It will be used to group all the shadow casters together. 
+This way we can show and hide them all at the same time.
+
+Each shadow caster is made of a :ref:`Sprite <class_Sprite>`, with a :ref:`LightOccluder2D <class_LightOccluder2D>` 
+child. For the demo the :ref:`Sprite <class_Sprite>` has a texture 
+set to the `caster image <https://raw.githubusercontent.com/godotengine/godot-demo-projects/master/2d/lights_and_shadows/caster.png>`_ and nothing else. The child :ref:`LightOccluder2D <class_LightOccluder2D>` is where all the magic happens. In a 
+game the :ref:`Sprite <class_Sprite>` could be more than a black box, it could be an image of whatever object is casting 
+the shadow, it could be a wall, a magical chest, or anything else.
+
+.. image:: img/light_shadow_sprites.png
+
+:ref:`LightOccluder2Ds <class_LightOccluder2D>` tell the game what the occluder looks like. They hold 
+an :ref:`OccluderPolygon2D <class_OccluderPolygon2D>` which is a container 
+for a polygon and some other information. For this demo, since our wall is a square. We 
+set ``Polygon`` to a square. The other default settings are fine.
+
+The first setting, ``Closed`` can be either ``on`` or ``off``. A closed polygon occludes light 
+coming from all directions. An open polygon only occludes light from only one direction
+
+``Cull Mode`` lets you select which direction gets culled. The default is ``Disabled``, meaning the occluder 
+will cast a shadow no matter which side the light is on. The other two settings ``Clockwise`` and 
+``Counter-Clockwise`` refer to the winding order of the vertices of the polygon. They essentially 
+pick out which direction will cast shadows. 
```

Comment:
I think this needs more explanation. Perhaps an image showing the different effects of the three settings.